packages/mobile/src/verify/reducer.ts

```diff
@@ -0,0 +1,340 @@
+        },
+      }
+    })
+    .addCase(setKomenciContext, (state, action) => {
+      return {
+        ...state,
+        komenci: {
+          ...state.komenci,
+          ...action.payload,
+        },
+      }
+    })
+    .addCase(setActionableAttestation, (state, action) => {
+      return {
+        ...state,
+        actionableAttestations: action.payload,
+      }
+    })
+    .addCase(setOverrideWithoutVerification, (state, action) => {
+      return {
+        ...state,
+        TEMPORAR_override_withoutVerification: action.payload,
+      }
+    })
+    .addCase(checkIfKomenciAvailable, (state) => {
+      return {
+        ...state,
+        komenciAvailable: undefined,
+      }
+    })
+    .addCase(setKomenciAvailable, (state, action) => {
+      return {
+        ...state,
+        komenciAvailable: action.payload,
+      }
+    })
+    .addCase(reset, (state, action) => {
+      return {
+        ...initialState,
+        e164Number: state.e164Number,
+        status: {
+          ...initialState.status,
+          komenci: action.payload.komenci,
+        },
+        komenciAvailable: action.payload.komenci,
+      }
+    })
+})
+
+const isBalanceSufficientForAttestations = (state: RootState, attestationsRemaining: number) => {
+  const userBalance = stableTokenBalanceSelector(state) || 0
+  return new BigNumber(userBalance).isGreaterThan(
+    attestationsRemaining * ESTIMATED_COST_PER_ATTESTATION
+  )
+}
+
+export const currentStateSelector = (state: RootState) => state.verify.currentState
+export const e164NumberSelector = (state: RootState) => state.verify.e164Number
+export const phoneHashSelector = (state: RootState) => state.verify.phoneHash
+export const komenciContextSelector = (state: RootState) => state.verify.komenci
+export const shouldUseKomenciSelector = (state: RootState) => {
+  if (state.verify.komenciAvailable === undefined) {
+    return undefined
+  }
+  const verificationHasStarted = state.verify.status.total > 0
+  // Only use Komenci when verification has not started with classic flow
+  return state.verify.komenciAvailable && !(verificationHasStarted && !state.verify.status.komenci)
+}
+
+export const verificationStatusSelector = (state: RootState) => state.verify.status
+export const actionableAttestationsSelector = (state: RootState): ActionableAttestation[] =>
+  state.verify.actionableAttestations
+export const overrideWithoutVerificationSelector = (state: RootState): boolean | undefined =>
+  state.verify.TEMPORAR_override_withoutVerification
+
+export const isBalanceSufficientForSigRetrievalSelector = (state: RootState) => {
+  const dollarBalance = stableTokenBalanceSelector(state) || 0
+  const celoBalance = celoTokenBalanceSelector(state) || 0
+  return isBalanceSufficientForSigRetrieval(dollarBalance, celoBalance)
+}
+
+// TODO: rewrite using reselect
+export const isBalanceSufficientSelector = (state: RootState) => {
+  const actionableAttestations = actionableAttestationsSelector(state)
```

Comment:
Do any of these selectors make network requests (ex. balance queries)? If so, we might get a small efficiency by calling `isBalanceSufficientSelector` only when `shouldUseKomenci` is false.

Reply:
It does not make a network request, selectors by definition should not do that, but thanks for pointing it out!

Reply:
Great, thanks for confirming!